src/azure-cli/azure/cli/command_modules/sql/custom.py

```diff
@@ -2196,6 +2193,47 @@ def managed_instance_encryption_protector_update(
         server_key_name=key_name
     )
 
+
+#####
+#           sql managed instance ad-admin
+#####
+
+
+def mi_ad_admin_set(
+        client,
+        resource_group_name,
+        managed_instance_name,
+        **kwargs):
+    '''
+    Creates a managed instance active directory administrator.
+    '''
+    from azure.cli.core._profile import Profile
+
+    profile = Profile()
+    sub = profile.get_subscription()
+    kwargs['tenant_id'] = sub['tenantId']
```

Comment:
Please extract function which gets tenant id so that the logic isn't duplicated between server and mi

Reply:
Done